diagnostic_aggregator/CMakeLists.txt

```diff
@@ -1,26 +1,65 @@
-catkin_package(DEPENDS diagnostic_msgs pluginlib roscpp rospy xmlrpcpp bondcpp
-    INCLUDE_DIRS include
-    LIBRARIES ${PROJECT_NAME})
-
-set(REQUIRED_diagnostic_msgs_VERSION_Indigo "1.11.9")
-set(REQUIRED_diagnostic_msgs_VERSION_Jade "1.12.4")
-if( ${diagnostic_msgs_VERSION} VERSION_LESS ${REQUIRED_diagnostic_msgs_VERSION_Indigo})
-  message(FATAL_ERROR "diagnostic_msgs version ${REQUIRED_diagnostic_msgs_VERSION_Indigo} or newer is required to build diagnotic_aggregator on ROS Indigo")
-endif()
-if( (${diagnostic_msgs_VERSION} VERSION_EQUAL "1.12.0") OR
-    (${diagnostic_msgs_VERSION} VERSION_GREATER "1.12.0" AND
-     ${diagnostic_msgs_VERSION} VERSION_LESS ${REQUIRED_diagnostic_msgs_VERSION_Jade}))
-  message(FATAL_ERROR "diagnostic_msgs version ${REQUIRED_diagnostic_msgs_VERSION_Jade} or newer is required to build diagnotic_aggregator on ROS Jade")
-endif()
-
-find_package(Boost REQUIRED COMPONENTS system)
-include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} gtest-1.7.0/include)
+#find_package(catkin REQUIRED diagnostic_msgs pluginlib roscpp rospy xmlrpcpp bondcpp)
+find_package(ament_cmake REQUIRED)
+find_package(builtin_interfaces REQUIRED)
+find_package(diagnostic_msgs REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclpy REQUIRED)
+find_package(xmlrpcpp REQUIRED)
+
+
+#catkin_package(DEPENDS diagnostic_msgs pluginlib roscpp rospy xmlrpcpp bondcpp
+#    INCLUDE_DIRS include
+#    LIBRARIES ${PROJECT_NAME})
+# Include boost
+find_package(Boost REQUIRED system)
+link_directories(${Boost_LIBRARY_DIRS})
+
+set(INCLUDE_DIRS
+  include
+  ${diagnostic_msgs_INCLUDE_DIRS}
+  ${builtin_interfaces_INCLUDE_DIRS}
+  ${rclcpp_INCLUDE_DIRS}
+  ${rclpy_INCLUDE_DIRS}
+  ${pluginlib_INCLUDE_DIRS}
+  ${xmlrpcpp_INCLUDE_DIRS}
+  ${Boost_INCLUDE_DIRS}
+)
+
+set(LIBS
+  ${diagnostic_msgs_LIBRARIES}
+  ${builtin_interfaces_LIBRARIES}
+  ${rclcpp_LIBRARIES}
+  ${rclpy_INCLUDE_DIRS}
+  ${pluginlib_INCLUDE_DIRS}
+  ${xmlrpcpp_INCLUDE_DIRS}
+  ${Boost_LIBRARIES}
+)
+
+#set(REQUIRED_diagnostic_msgs_VERSION_Indigo "1.11.9")
+#set(REQUIRED_diagnostic_msgs_VERSION_Jade "1.12.4")
+#if( ${diagnostic_msgs_VERSION} VERSION_LESS ${REQUIRED_diagnostic_msgs_VERSION_Indigo})
+#  message(FATAL_ERROR "diagnostic_msgs version ${REQUIRED_diagnostic_msgs_VERSION_Indigo} or newer is required to build diagnotic_aggregator #on ROS Indigo")
+#endif()
+#if( (${diagnostic_msgs_VERSION} VERSION_EQUAL "1.12.0") OR
+#    (${diagnostic_msgs_VERSION} VERSION_GREATER "1.12.0" AND
+#     ${diagnostic_msgs_VERSION} VERSION_LESS ${REQUIRED_diagnostic_msgs_VERSION_Jade}))
+#  message(FATAL_ERROR "diagnostic_msgs version ${REQUIRED_diagnostic_msgs_VERSION_Jade} or newer is required to build diagnotic_aggregator on #ROS Jade")
+#endif()
+
+#find_package(Boost REQUIRED COMPONENTS system)
+
+#include_directories(include ${catkin_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS} gtest-1.7.0/include)
```

Comment:
Dead code. Remove.

Reply:
Dead code is removed in new pull request